plugin/evm/vm.go

```diff
@@ -210,7 +229,9 @@ func init() {
 
 // VM implements the snowman.ChainVM interface
 type VM struct {
-	ctx *snow.Context
+	ctx           *snow.Context   // TODO rename to snowCtx
+	backgroundCtx context.Context // TODO rename to ctx
+	cancel        context.CancelFunc
```

Comment:
Why are we embedding `context.Context` in the VM? This is explicitly recommended against in golang.

Reply:
This is because the gossip package uses contexts to cancel (e.g `gossiper.Gossip(ctx)` so the caller can shut down the gossip thread by calling `cancel()`. The alternative is to use a channel

Reply:
Imo we should not embed a context since there's a very direct recommendation against this.

Could we use the alternative?

Reply:
This should be safe since we're only using the contexts to to cancel the goroutine, we can definitely use a channel but since it requires us to keep going back-and-forth between avalanchego and coreth to cut more rcs I think it would be easier to do this as a follow-up to this PR

Reply:
Discussed offline, this is fine since we don't include key-value pairs in contexts here, so will need to be aware of how this is used going forward.

Reply:
There is really no reason why this context is stored on the VM. We only need `cancel`. The context can be left as a local variable here.